wpilibj/src/main/java/edu/wpi/first/wpilibj/ResistanceCalculator.java

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.wpilibj;
+
+import edu.wpi.first.util.CircularBuffer;
+
+/**
+ * Finds the resistance of a channel using a running linear regression over a window. Must be
+ * updated with current and voltage periodically using the {@link
+ * ResistanceCalculator#calculate(double, double) calculate} method.
+ *
+ * <p>To use this for finding the resistance of a channel, use the calculate method with the battery
+ * voltage minus the voltage at the motor controller or whatever is plugged in to the PDP at that
+ * channel.
+ */
+public class ResistanceCalculator {
+  public static final int kDefaultBufferSize = 250;
+  public static final double kDefaultRSquaredThreshold = 0.75;
```

Comment:
The default in the C++ version is 0.5. Should this be 0.5 or 0.75?

Reply:
Good catch, it should be 0.75 in both.